src/app/project_runner.go

```diff
@@ -83,7 +85,25 @@ func (p *ProjectRunner) runProcess(config *types.ProcessConfig) {
 		procLog = pclog.NewLogBuffer(0)
 	}
 	procState, _ := p.GetProcessState(config.ReplicaName)
-	process := NewProcess(p.project.Environment, procLogger, config, procState, procLog, *p.project.ShellConfig)
+	isMain := config.Name == p.mainProcess
+	hasMain := p.mainProcess != ""
+	printLogs := !hasMain
+	extraArgs := []string{}
+	if isMain {
+		extraArgs = p.mainProcessArgs
+		config.RestartPolicy.ExitOnEnd = true
```

Comment:
since the purpose of `run` is to run a single process and it's deps. Once the main process exits there doesn't seem to be any reason to keep the dependencies around.

Reply:
Don't you think that it should be up to the user to decide that?
Aren't there use cases where the rest of the apps should continue running, even after the tui app exits?

Reply:
`run` command doesn't start a TUI, it only works in command line mode. We attach stdio to the main process so that it can be interacted with and we only show the logs for that process. When the process end, we exit PC with the same code as the main process.

Note that you can still attach a TUI to PC which is started via the `run` command, but since interactivity is not implemented in the TUI, starting in TUI mode via `run` would be no different then running `up main-process-name`

Given the above, I think it would be counter intuitive to keep supporting services running once the main process exits as all IO would stop, and the program would simply silently hang there, without passing control back to the shell's command line. 

If you feel strongly about giving users the choice to do, I would suggest keeping current behaviour as default and allow it to be overridden via `--exit-on-end=false`.

But TBH, I would rather wait until somebody actually comes up with a valid use case for it.

Reply:
CLI Applications like `gdb --tui` is a HUGE added bonus of your PR. I know that it wasn't intended to run TUI, but without your feature it wouldn't be possible in Process Compose 😃

Let's go with the "wait and see" approach that you proposed. If someone will ask for it we can always add another flag (`--exit-on-end=false`)